### PR TITLE
Ensuring LPIPS model is always frozen and in eval mode

### PR DESCRIPTION
**Pull request by jordan-benjamin**
This PR modifies the LPIPS metric so that the model always stays in eval mode and all of the model's parameters are non-trainable/frozen.

I noticed when I was using LPIPS as a perceptual loss in PyTorch Lightning with `track_grad_norm` on that the gradients of the params of the LPIPS model were being logged, when really they should not be receiving gradients at all. This is because:

- If someone used the LPIPS metric (which itself is a `nn.Module` subclass) as a subcomponent of a larger model and passed the bigger model's `.parameters()` to an optimizer, the LPIPS model parameters would update. 
- Depending on the experiment setup, gradients might be needlessly calculated for the LPIPS model parameters, wasting compute + memory. 

This PR freezes all the LPIPS model parameters in both the functional and module metric APIs. I also put the model into eval mode, and for the metric API ensure that training mode can never be turned on, following the [torchmetrics implementation of FID](https://github.com/PyTorchLightning/metrics/blob/484477a24a22b18315486da6be65eed5135e39f1/torchmetrics/image/fid.py#L38).

### CODE REVIEW

**Review by mannatsingh — APPROVED** [review]
The model creation and conversion to eval mode seems to be repeated in the two files. I'm not sure if we should move it to a function or keep it this way. Your call!

Currently the diff includes the FLOP changes too, just make sure to strip them before landing. Also, it'd be great if Matt also reviewed this :)

**Review by mmuckley — APPROVED** [review]
I echo @mannatsingh's recommendations. Make sure you remove the flop counting notebook from the PR.

**Pull request merged**
